src/deck.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * @copyright Copyright (c) 2020 Vincent Petry <vincent@nextcloud.com>
+ *
+ * @author Vincent Petry <vincent@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import Vue from 'vue'
+import { generateFilePath, generateOcsUrl, generateUrl } from '@nextcloud/router'
+import { getRequestToken } from '@nextcloud/auth'
+import { translate, translatePlural } from '@nextcloud/l10n'
+import { showSuccess, showError } from '@nextcloud/dialogs'
+import SHA1 from 'crypto-js/sha1'
+import Hex from 'crypto-js/enc-hex'
+import axios from '@nextcloud/axios'
+
+// CSP config for webpack dynamic chunk loading
+// eslint-disable-next-line
+__webpack_nonce__ = btoa(getRequestToken())
+
+// Correct the root of the app for chunk loading
+// OC.linkTo matches the apps folders
+// OC.generateUrl ensure the index.php (or not)
+// We do not want the index.php since we're loading files
+// eslint-disable-next-line
+__webpack_public_path__ = generateFilePath('spreed', '', 'js/')
+
+Vue.prototype.t = translate
+Vue.prototype.n = translatePlural
+Vue.prototype.OC = OC
+Vue.prototype.OCA = OCA
+
+document.addEventListener('DOMContentLoaded', function() {
+
+	if (!window.OCA.Deck) {
+		return
+	}
+
+	window.OCA.Deck.registerCardAction({
+		label: t('spreed', 'Link to a conversation'),
+		icon: 'icon-talk',
+		callback: (card) => {
+			OCP.Collaboration.trigger('room').then(async(token) => {
+				const tempId = 'temp-' + card.id + '-' + token + '-' + (new Date().getTime())
+				try {
+					const response = await axios.post(generateOcsUrl('apps/spreed/api/v1', 2) + `chat/${token}/share`, {
+						objectType: 'deck-card',
+						objectId: card.id,
+						metaData: JSON.stringify(card),
+						referenceId: Hex.stringify(SHA1(tempId)),
+					})
+					const messageId = response.data.ocs.data.id
+					const targetUrl = generateUrl('/call/{token}#message_{messageId}', { token, messageId })
+					showSuccess(t('spreed', 'Deck card has been linked with the selected <a href="{link}">conversation</a>.', {
```

Comment:
We don't have the conversation name, do we?

Reply:
```suggestion
					showSuccess(t('spreed', 'Deck card has been posted with the selected <a href="{link}">conversation</a>.', {
```

Reply:
> We don't have the conversation name, do we?

We only have the token. But we could replace it with the full conversation object.
However we can only afford to do this if we're sure that the RoomSelector is not used elsewhere outside of Talk, otherwise it would be an API change...